src/nvme/tree.c

```diff
@@ -725,8 +725,8 @@ static int nvme_scan_subsystem(struct nvme_root *r, const char *name,
 			if (strcmp(_s->name, name))
 				continue;
 			if (!__nvme_scan_subsystem(r, _s, f, f_args)) {
-				errno = -EINVAL;
-				goto out_free;
+				errno = EINVAL;
+				return -1;
```

Comment:
The `errno` change could properly be it's own patch as it fixes an existing bug. I suppose the `errno` should have been `ret`. But then it's just very minor thing and I am fine as it is. It's something I stumbled upon while reading through the change.

Reply:
Sure, I can break that out into a separate patch. Looks like the code meant to set `ret = -EINVAL` instead of `errno = -EINVAL`.